source/includes/steps-fle-convert-to-a-remote-master-key-kmip.yaml

```diff
@@ -0,0 +1,132 @@
+           -Djavax.net.ssl.keyStorePassword=<KeyStore password>
+           -Djavax.net.ssl.trustStoreType=jks
+           -Djavax.net.ssl.trustStore=<path to TrustStore containing KMIP KMS CA file>
+           -Djavax.net.ssl.trustStorePassword=<TrustStore password>
+      
+        .. note:: Configure Client With SSLContext
+
+           If you would rather configure your KMIP client using an SSL context, use the 
+           `kmsProviderSslContextMap <{+java-api+}/apidocs/mongodb-driver-core/com/mongodb/ClientEncryptionSettings.Builder.html#kmsProviderSslContextMap(java.util.Map)>`__
+           method.
+
+---
+title: Create a New Data Encryption Key
+ref: create-a-new-data-key
+content: |
+  To encrypt your data, you need a data encryption key generated from your
+  KMS-hosted **master key**. The following diagram shows the requests you need
+  to make from the client application to create and store a new **data
+  encryption key**:
+
+  .. image:: /figures/CSFLE_Data_Key_KMS.png
+     :alt: Diagram that describes creating a data encryption key when using a KMS provider
+
+  1. First, specify the following information to access the master key:
+
+     .. list-table::
+        :header-rows: 1
+        :stub-columns: 1
+        :widths: 30 15 45
+
+        * - Field
+          - Required
+          - Description
+
+        * - keyId
+          - No
+          - The ``keyId`` field of a 96 byte
+            `Secret Data managed object <http://docs.oasis-open.org/kmip/spec/v1.4/os/kmip-spec-v1.4-os.html#_Toc490660780>`__
+            stored in your KMIP KMS.
+
+            .. note:: Create a New Master Key
+
+               If you do not specify the ``keyId`` field in the ``masterKey`` document
+               you send to your KMIP KMS, the driver creates a new
+               96 Byte Secret Data managed object in your KMS to act as your
+               master key. 
+
+        * - endpoint
+          - Yes
+          - The URI of your KMIP KMS.
+
+  2. Once you have the required information, update and run the following code
+     to generate the new data encryption key:
+     
+  .. include:: /includes/substitute-placeholders.rst
+
+  .. tabs-drivers::
+
+     .. tab::
+        :tabid: java-sync
+
+        .. code-block:: java
+
+           Map<String, Map<String, Object>> kmsProviderProperties = new HashMap<>();
+           Map<String, Object> providerDetails = new HashMap<>();
+           providerDetails.put("endpoint", "<KMIP KMS URI>");
+           kmsProviderProperties.put(kmsProvider,  providerDetails);
+           String keyVaultCollection = "<MongoDB namespace where you store your keys>"
+
+           ClientEncryption clientEncryption = ClientEncryptions.create(ClientEncryptionSettings.builder()
+               .keyVaultMongoClientSettings(MongoClientSettings.builder()
+                   .applyConnectionString(new ConnectionString("mongodb://localhost:27017"))
```

Comment:
Perhaps: `new ConnectionString("<MongoDB connection string"> ?

Reply:
I agree, "<MongoDB Connection String>" is more helpful here. It seems to be an issue throughout the CSFLE guide that the "mongodb://localhost:27017" URI is hardcoded into code snippets rather than providing an angle-bracketed string readers should replace. I've updated this instance and created [a ticket to fix the other connection URIs](https://jira.mongodb.org/browse/DOCSP-20106) in this guide.